drivers/platform/maxim/max32655/maxim_spi.c

```diff
@@ -78,25 +78,31 @@ static void _max_delay_config(struct no_os_spi_desc *desc,
 	mxc_spi_regs_t *spi;
 	uint32_t clk_rate;
 	uint32_t ticks_ns;
+	uint32_t delay_first_ns;
+	uint32_t delay_last_ns;
 
-	if (msg->cs_delay_first == st->cs_delay_first &&
-	    msg->cs_delay_last == st->cs_delay_last)
+	delay_first_ns = msg->cs_delay_first * 1000 +
```

Comment:
You seem to be using scenario A, there is also scenario B. Not sure which one is best.
B would result in less and simpler code.
```
                                 ┌──────────────────┐
                                 │    user          │
                                 └───────┬──────────┘
                                         │
                                         │
                                         │ use driver to get some data from ad1234
                                         │
                                         │
                                         │
                                         │
                                ┌────────▼───────────┐
                                │ ad1234 driver      │
                                └────────┬───────────┘
                                         │
                                         │
                                         │configure delays as in datasheet of ad1234
                                         │
                                         │
                                         │
                                 ┌───────▼───────────┐
                                 │ no_os_spi         │
                                 └───────┬───────────┘
                       A                 │               B
                                         │
                  add the above delays   │either use the above delays
to the board specific delays that may or │or override above delays with board specific delays
may not have been provided by the user   │if they were provided by the user at initialization
at initialization                        │
                                 ┌───────▼───────────┐
                                 │spi peripheral     │
                                 └───────────────────┘
```